sdk/core/Azure.Core.TestFramework/src/TestProxy.cs

```diff
@@ -0,0 +1,218 @@
+                ImportDevCertIfNeeded();
+            }
+
+            _testProxyProcess = Process.Start(testProxyProcessInfo);
+            ProcessTracker.Add(_testProxyProcess);
+            _ = Task.Run(
+                () =>
+                {
+                    while (!_testProxyProcess.HasExited && !_testProxyProcess.StandardError.EndOfStream)
+                    {
+                        var error = _testProxyProcess.StandardError.ReadLine();
+                        throw new TestProxyException($"An error occurred in the test proxy: {error}");
+                    }
+                });
+
+            int lines = 0;
+            while ((_proxyPortHttp == null || _proxyPortHttps == null) && lines++ < 20)
+            {
+                string outputLine = _testProxyProcess.StandardOutput.ReadLine();
+                // useful for debugging
+                TestContext.Progress.WriteLine(outputLine);
+
+                if (ProxyPortHttp == null && TryParsePort(outputLine, "http", out _proxyPortHttp))
+                {
+                    continue;
+                }
+
+                if (_proxyPortHttps == null && TryParsePort(outputLine, "https", out _proxyPortHttps))
+                {
+                    continue;
+                }
+            }
+
+            if (_proxyPortHttp == null || _proxyPortHttps == null)
+            {
+                throw new TestProxyException("Failed to start the test proxy.");
+            }
+
+            // we need to use https when talking to test proxy admin endpoint so that we can establish the connection before any
+            // test related traffic happens which would send a Connection header for the first request. This can be switched to HTTP
+            // once https://github.com/Azure/azure-sdk-tools/issues/2303 is fixed
+            var handler = new HttpClientHandler
+            {
+                ServerCertificateCustomValidationCallback = (_, certificate, _, _) => certificate.Issuer == TestProxy.DevCertIssuer
+            };
+            var options = new TestProxyClientOptions
+            {
+                Transport = new HttpClientTransport(handler)
+            };
+            Client = new TestProxyRestClient(
+                new ClientDiagnostics(options),
+                HttpPipelineBuilder.Build(options),
+                new Uri($"https://{IpAddress}:{_proxyPortHttps}"));
+
+            // For some reason draining the standard output stream is necessary to keep the test-proxy process healthy. Otherwise requests
+            // start timing out. This only seems to happen when not specifying a port.
+            _ = Task.Run(
+                () =>
+                {
+                    while (!_testProxyProcess.HasExited && !_testProxyProcess.StandardOutput.EndOfStream)
+                    {
+                        _testProxyProcess.StandardOutput.ReadLine();
+                    }
+                });
+        }
+
+        public static TestProxy Start()
```

Comment:
Inline this into the default ctor implementation?

Reply:
I wanted to have the method because it is doing a lot more work than a typical constructor.